pkg/services/librarypanels/librarypanels.go

```diff
@@ -39,6 +44,63 @@ func (lps *LibraryPanelService) IsEnabled() bool {
 	return lps.Cfg.IsPanelLibraryEnabled()
 }
 
+// LoadLibraryPanelsForDashboard loads library panels JSON from the database for a dashboard.
+func (lps *LibraryPanelService) LoadLibraryPanelsForDashboard(dash *models.Dashboard) error {
+	if !lps.IsEnabled() {
+		return nil
+	}
+
+	libraryPanels, err := lps.getLibraryPanelsForDashboardID(dash.Id)
+	if err != nil {
+		return err
+	}
+
+	panels := dash.Data.Get("panels").MustArray()
+	for i, panel := range panels {
+		panelAsJSON := simplejson.NewFromAny(panel)
+		libraryPanel := panelAsJSON.Get("libraryPanel")
+		if libraryPanel.Interface() == nil {
+			continue
+		}
+
+		// we have a library panel
+		uid := libraryPanel.Get("uid").MustString()
+		if len(uid) == 0 {
+			return errors.New("found a library panel without uid")
+		}
+
+		libraryPanelInDB, ok := libraryPanels[uid]
+		if !ok {
+			return errors.New("found a library panel that does not exists as a connection")
+		}
+
+		// we have a match between what is stored in db and in dashboard json
+		libraryPanelModel, err := libraryPanelInDB.Model.MarshalJSON()
+		if err != nil {
+			return fmt.Errorf("could not marshal library panel JSON: %w", err)
+		}
+
+		libraryPanelModelAsJSON, err := simplejson.NewJson(libraryPanelModel)
+		if err != nil {
+			return fmt.Errorf("could not convert library panel to JSON: %w", err)
```

Comment:
I realize that this is converting what is already JSON to simplejson, right?
```suggestion
			return fmt.Errorf("could not convert library panel to simplejson model: %w", err)
```

Reply:
You're correct